src/commands/data/import/resume.ts

```diff
@@ -0,0 +1,193 @@
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { SfCommand, Flags } from '@salesforce/sf-plugins-core';
+import { Messages } from '@salesforce/core';
+import { MultiStageOutput } from '@oclif/multi-stage-output';
+import terminalLink from 'terminal-link';
+import { JobInfoV2 } from '@jsforce/jsforce-node/lib/api/bulk2.js';
+import { BulkImportRequestCache } from '../../../bulkDataRequestCache.js';
+
+Messages.importMessagesDirectoryFromMetaUrl(import.meta.url);
+const messages = Messages.loadMessages('@salesforce/plugin-data', 'data.import.resume');
+
+export type DataImportResumeResult = {
+  jobId: string;
+  processedRecords?: number;
+  successfulRecords?: number;
+  failedRecords?: number;
+};
+
+export default class DataImportResume extends SfCommand<DataImportResumeResult> {
+  public static readonly summary = messages.getMessage('summary');
+  public static readonly description = messages.getMessage('description');
+  public static readonly examples = messages.getMessages('examples');
+
+  public static readonly flags = {
+    'use-most-recent': Flags.boolean({
+      summary: messages.getMessage('flags.use-most-recent.summary'),
+      exactlyOne: ['job-id'],
+    }),
+    'job-id': Flags.salesforceId({
+      summary: messages.getMessage('flags.job-id.summary'),
+      char: 'i',
+      length: 18,
+      startsWith: '750',
+      exactlyOne: ['use-most-recent'],
+    }),
+    wait: Flags.duration({
+      char: 'w',
+      unit: 'minutes',
+      summary: messages.getMessage('flags.wait.summary'),
+      defaultValue: 5,
+    }),
+  };
+
+  public async run(): Promise<DataImportResumeResult> {
+    const { flags } = await this.parse(DataImportResume);
+
+    const cache = await BulkImportRequestCache.create();
+
+    const resumeOpts = await cache.resolveResumeOptionsFromCache(
+      flags['job-id'],
+      flags['use-most-recent'],
+      undefined,
+      undefined
+    );
+
+    const conn = resumeOpts.options.connection;
+
+    const ms = new MultiStageOutput<JobInfoV2>({
+      jsonEnabled: flags.json ?? false,
+      stages: ['Creating ingest job', 'Processing the job'],
+      title: 'Importing data',
+      stageSpecificBlock: [
+        {
+          stage: 'Processing the job',
+          label: 'Processed records',
+          type: 'dynamic-key-value',
+          get: (data): string | undefined => {
+            if (data?.numberRecordsProcessed) {
+              return data.numberRecordsProcessed.toString();
+            }
+          },
+        },
+        {
+          stage: 'Processing the job',
+          label: 'Successful records',
+          type: 'dynamic-key-value',
+          get: (data): string | undefined => {
+            const numberRecordsFailed = data?.numberRecordsFailed ?? 0;
+
+            if (data?.numberRecordsProcessed) {
+              return (data.numberRecordsProcessed - numberRecordsFailed).toString();
```

Comment:
we default to `0` on L84 for failed records if no API info is available yet (this state update happens while job is being processed) but only render the `Successful records` block  if the API returns `numberRecordsProcessed`.
This makes oclif/mso render a spinner instead of `0` when there's no processed data (first seconds of a job run or if it failed).